src/main/java/com/socialmedia/security/JWTAuthenticationFilter.java

```diff
@@ -0,0 +1,57 @@
+package com.socialmedia.security;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+
+import static com.socialmedia.security.SecurityConstants.HEADER_STRING;
+import static com.socialmedia.security.SecurityConstants.TOKEN_PREFIX;
+import static com.socialmedia.security.SecurityConstants.SECRET;
+
+
+public class JWTAuthenticationFilter extends BasicAuthenticationFilter {
+
+  public JWTAuthenticationFilter(AuthenticationManager authenticationManager) {
+    super(authenticationManager);
+  }
+
+  @Override
+  protected void doFilterInternal(HttpServletRequest req,
+                                  HttpServletResponse res,
+                                  FilterChain chain) throws IOException, ServletException {
+
+    String header = req.getHeader(HEADER_STRING);
+
+    if (header == null || !header.startsWith(TOKEN_PREFIX)) {
+      chain.doFilter(req, res);
+      return;
```

Comment:
I don't like this empty return... doFilterInternal method is void, you may set authentication for reversed condition - header !=null && header.startsWith(TOKEN_PREFIX)

Reply:
done